engines/e_devcrypto.c

```diff
@@ -34,6 +34,14 @@
 
 #define engine_devcrypto_id "devcrypto"
 
+/*
+ * Use session2_op on FreeBSD which permits requesting specific
+ * drivers or classes of drivers at session creation time.
+ */
+#ifdef CIOCGSESSION2
```

Comment:
This is a bit gross.  However, using an #ifdef in all the places that use `struct session_op` is also gross.  Another alternative might be to define a local typedef (e.g. `session_op_t`) that is defined to either `struct session_op` or `struct session2_op`

Reply:
Yeah, I didn't think of the typedef until I was typing this comment after uploading the request.  I've changed it to use a typedef.